src/python/pants_release/generate_github_workflows.py

```diff
@@ -867,9 +847,35 @@ def build_wheels_job(
             "steps": [
                 *initial_steps,
                 *([] if platform == Platform.LINUX_ARM64 else [install_go()]),
-                *helper.build_wheels(),
+                {
+                    "name": "Build wheels",
+                    "run": "./pants run src/python/pants_release/release.py -- build-wheels",
+                    "env": helper.platform_env(),
+                },
                 helper.upload_log_artifacts(name="wheels"),
                 *([deploy_to_s3("Deploy wheels to S3")] if for_deploy_ref else []),
+                {
+                    "name": "Build Pants PEX",
+                    "run": "./pants package src/python/pants:pants-pex",
+                    "env": helper.platform_env(),
+                },
+                *(
+                    [
+                        {
+                            "name": "Upload Pants PEX",
+                            "if": "needs.release_info.outputs.is-release == 'true'",
+                            "run": dedent(
+                                """\
+                                LOCAL_TAG=$(PEX_INTERPRETER=1 dist/src.python.pants/pants-pex.pex -c "import sys;major, minor = sys.version_info[:2];import os;uname = os.uname();print(f'cp{major}{minor}-{uname.sysname.lower()}_{uname.machine.lower()}')")
```

Comment:
This is my assume-nothing way of generating the tag. Using the PEX itself means we'll land the right IC, and using Python means we'll be confident `scie-pants`'s Python script can produce a similar tag.

The value looks like: `cp39_linux_x86_64`